src/lib.rs

```diff
@@ -108,8 +108,13 @@ fn parse_snippet(span: &DiagnosticSpan) -> Option<Snippet> {
     let text_slice = span.text[0].text.chars().collect::<Vec<char>>();
 
     // We subtract `1` because these highlights are 1-based
-    let start = span.text[0].highlight_start - 1;
-    let end = span.text[0].highlight_end - 1;
+    // Check the `min` so that it doesn't attempt to index out-of-bounds when
+    // the span points to the "end" of the line. For example, a line of
+    // "foo\n" with a highlight_start of 5 is intended to highlight *after*
+    // the line. This needs to compensate since the newline has been removed
+    // from the text slice.
```

Comment:
Oh? Is this a change that the newlines is not part of the text slice?

Reply:
AFAIK, rustc has never included line endings in the "text" lines.

It is somewhat unusual for spans to point past the end of the line. `no_main` is one of the few ways to do it.  The `unclosed delimiter` error is the only other way I know of.